kong/db/declarative/init.lua

```diff
@@ -802,6 +802,38 @@ function declarative.load_into_cache_with_events(entities, meta, hash)
     return nil, "failed to persist cache page number inside shdict: " .. err
   end
 
+  local worker_count = ngx.worker.count()
+  local sleep_left = 60
+  local sleep_time = 0.0375
+
+  while true do
+    local flips = ngx.shared.kong:get("declarative:flips")
+    if flips == nil then
+      break
+    end
+
+    if flips == worker_count then
+      break
+    end
+
+    if sleep_left == 0 then
```

Comment:
We can remove this and move it to the `while` condition as commented above.

Also, from a defensive programming perspective, we usually want to use `sleep_left <= 0` instead of `==`.

Reply:
the code below does not allow it to go negative.